File: 2025-10-07 - Padroes de Projetos - Iterator/Exemplo_Python/Livro.py
class Livro:
   def __init__(self, titulo, autor, genero):
      self.__titulo = titulo
      self.__autor = autor
      self.__genero = genero

   # Getters e Setters
   def getTitulo(self):
      return self.__titulo

   def setTitulo(self, titulo):
      self.__titulo = titulo

   def getAutor(self):
      return self.__autor

   def setAutor(self, autor):
      self.__autor = autor

   def getGenero(self):
      return self.__genero

   def setGenero(self, genero):
      self.__genero = genero

   def __str__(self):
      return f"Livro: {self.__titulo} | Autor: {self.__autor} | Gênero: {self.__genero}"

File: 2025-10-07 - Padroes de Projetos - Iterator/Exemplo_C#/Livro.cs
using System;

public class Livro {
   private string titulo;
   private string autor;
   private string genero;

   public Livro(string titulo, string autor, string genero) {
      this.titulo = titulo;
      this.autor = autor;
      this.genero = genero;
   }

   public string Titulo {
      get { return titulo; }
      set { titulo = value; }
   }

   public string Autor {
      get { return autor; }
      set { autor = value; }
   }

   public string Genero {
      get { return genero; }
      set { genero = value; }
   }

   public override string ToString() {
      return $"Livro: {titulo} | Autor: {autor} | Gênero: {genero}";
   }
}
